src/test/java/com/wikia/webdriver/elements/mercury/components/Navigation.java

```diff
@@ -178,4 +116,49 @@ public Navigation openRecentWikiActivity() {
 
     return this;
   }
+
+  public boolean isUserProfileLinkVisible() {
+    try {
+      wait.forElementVisible(userProfile);
+      PageObjectLogging.logInfo("User profile link is visible");
+      return true;
+    } catch (Exception e) {
+      PageObjectLogging.logInfo("User profile link is not visible");
+      return false;
+    }
+  }
+
+
+  public boolean isMainHeaderVisible() {
+    try {
+      wait.forElementVisible(navigationMainHeader);
+      PageObjectLogging.logInfo("Main header is visible");
+      return true;
+    } catch (Exception e) {
+      PageObjectLogging.logInfo("Main header is not visible");
+      return false;
+    }
+  }
+
+  public boolean isBackButtonVisible() {
+    try {
+      wait.forElementVisible(backButton);
+      PageObjectLogging.logInfo("Back button is visible");
+      return true;
+    } catch (Exception e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)